src/edit_mode/vi/parser.rs

```diff
@@ -108,13 +108,19 @@ impl ParsedViSequence {
             | (Some(Command::RewriteCurrentLine), ParseResult::Incomplete)
             | (Some(Command::SubstituteCharWithInsert), ParseResult::Incomplete)
             | (Some(Command::HistorySearch), ParseResult::Incomplete)
-            | (Some(Command::Change), ParseResult::Valid(_)) => Some(ViMode::Insert),
+            | (Some(Command::Change), ParseResult::Valid(_))
+            | (Some(Command::Change), ParseResult::Incomplete) => Some(ViMode::Insert),
```

Comment:
Not sure if the mode change should occur for the `Change`/`Incomplete` combination already? At least in normal mode we should remain there. Different story in Visual mode, so maybe that is what you are trying to achieve. 

We just may want to defend against it becoming an issue in normal mode.

Same for the delete part.